src/cpp/src/continuous_batching/pipeline_base.cpp

```diff
@@ -278,6 +279,12 @@ ContinuousBatchingPipeline::IContinuousBatchingPipeline::generate(
         std::string templated_history = m_tokenizer.apply_chat_template(m_history, true);
 
         m_inputs_embedder->set_apply_chat_template_status(false);
+
+        if (sampling_params[0].is_prompt_lookup()) {
+            auto prompt_ids = m_inputs_embedder->encode_prompt(prompt);
```

Comment:
Why not `unified_prompt` used? Inside inputs embedder `unified_prompt` is encoded, not the original prompt, so I suppose here we should do the same.